op-node/p2p/sync_test.go

```diff
@@ -251,7 +269,15 @@ func TestMultiPeerSync(t *testing.T) {
 	// But the re-request checks the status in the main loop, and it may thus look like it's still in-flight,
 	// and thus not run the new request.
 	// Wait till the failed request is recognized as marked as done, so the re-request actually runs.
-	for !clB.inFlight[25].Load() {
+	ctx, cancelFunc = context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancelFunc()
+	for clB.inFlight[25] != nil && !clB.inFlight[25].Load() {
+		select {
+		case <-ctx.Done():
+			t.Fatal("Did not complete request for block 25 in timely manner")
+		default:
+			// Carry on, timeout not yet reached
+		}
```

Comment:
Part of the problem is that the original fix of mine was wrong here, and we shouldn't read from `inFlight` map at all here, because it can be concurrently changed

Reply:
hmm, good point - while our code can handle the race conditions involved with two threads modifying the map, I suspect go's map implementation may not like it.

I've rebased the original commit onto latest develop and pushed up an extra commit that changes it to use a new `isInFlight` method on `SyncClient`. It means adding new production code but I think it makes sense and its the least intrusive thing I could think of to be able to know that the response has actually been processed.